File: Solutions/하루 한문제 코테준비/성격유형검사하기.py
from collections import defaultdict
def solution(survey, choices):
    answer = ''
    personality = defaultdict(int)
    each_personal = [['R','T'],['C','F'],['J','M'],['A','N']]

    for i in range(len(survey)):
        [disagree, agree] = list(survey[i])
        num = choices[i]

        if(4-num > 0):
            personality[disagree] += abs(4-num)
        else:
            personality[agree] += abs(4-num)

    for first,second in each_personal:
        if(personality[first]<personality[second]):
            answer += second
        else:
            answer += first
    return answer

print(solution(["AN", "CF", "MJ", "RT", "NA"], [5, 3, 2, 7, 5]))

File: Solutions/하루 한문제 코테준비/javascript_solve/성격유형검사하기.js
function solution(survey, choices) {
  var answer = "";
  const eachPersonal = [
    ["R", "T"],
    ["C", "F"],
    ["J", "M"],
    ["A", "N"],
  ];
  const result = { R: 0, T: 0, C: 0, F: 0, J: 0, M: 0, A: 0, N: 0 };

  survey.forEach((val, idx) => {
    const [first, second] = val.split("");
    // console.log(first, second);
    if (choices[idx] > 4) {
      result[`${second}`] += choices[idx] - 4;
    } else if (choices[idx] < 4) {
      result[`${first}`] += 4 - choices[idx];
    }
  });
//   console.log(result);
  for (let i = 0; i < eachPersonal.length; i++) {
    const [first, second] = eachPersonal[i];
    if (result[`${first}`] < result[`${second}`]) {
      answer += `${second}`;
    } else {
      answer += `${first}`;
    }
  }
  return answer;
}
console.log(solution(["AN", "CF", "MJ", "RT", "NA"], [5, 3, 2, 7, 5]));
